src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/BrowserWebSockets/ClientWebSocketOptions.cs

```diff
@@ -0,0 +1,122 @@
+            throw new PlatformNotSupportedException();
+        }
+
+        public bool UseDefaultCredentials
+        {
+            get => throw new PlatformNotSupportedException();
+            set => throw new PlatformNotSupportedException();
+        }
+
+        public System.Net.ICredentials Credentials
+        {
+            get => throw new PlatformNotSupportedException();
+            set => throw new PlatformNotSupportedException();
+        }
+
+        public System.Net.IWebProxy Proxy
+        {
+            get => throw new PlatformNotSupportedException();
+            set => throw new PlatformNotSupportedException();
+        }
+
+        public X509CertificateCollection ClientCertificates
+        {
+            get => throw new PlatformNotSupportedException();
+            set => throw new PlatformNotSupportedException();
+        }
+
+        public System.Net.Security.RemoteCertificateValidationCallback RemoteCertificateValidationCallback
+        {
+            get => throw new PlatformNotSupportedException();
+            set => throw new PlatformNotSupportedException();
+        }
+
+        public System.Net.CookieContainer Cookies
+        {
+            get => throw new PlatformNotSupportedException();
+            set => throw new PlatformNotSupportedException();
+        }
+
+        #endregion HTTP Settings
+
+        #region WebSocket Settings
+
+        public void AddSubProtocol(string subProtocol)
+        {
+            ThrowIfReadOnly();
+            WebSocketValidate.ValidateSubprotocol(subProtocol);
+
+            // Duplicates not allowed.
+            List<string> subprotocols = RequestedSubProtocols; // force initialization of the list
+            foreach (string item in subprotocols)
+            {
+                if (string.Equals(item, subProtocol, StringComparison.OrdinalIgnoreCase))
+                {
+                    throw new ArgumentException(SR.Format(SR.net_WebSockets_NoDuplicateProtocol, subProtocol), nameof(subProtocol));
+                }
+            }
+            subprotocols.Add(subProtocol);
+        }
+
+        internal List<string> RequestedSubProtocols => _requestedSubProtocols ??= new List<string>();
+
+        public TimeSpan KeepAliveInterval
+        {
+            get => throw new PlatformNotSupportedException();
+            set => throw new PlatformNotSupportedException();
+        }
+
+        public void SetBuffer(int receiveBufferSize, int sendBufferSize)
+        {
+            throw new NotImplementedException();
+        }
+
+        public void SetBuffer(int receiveBufferSize, int sendBufferSize, ArraySegment<byte> buffer)
+        {
+            throw new PlatformNotSupportedException();
+        }
+
+        #endregion WebSocket settings
+
+        #region Helpers
+
+        internal void SetToReadOnly()
+        {
+            Debug.Assert(!_isReadOnly, "Already set");
+            _isReadOnly = true;
+        }
+
+        private void ThrowIfReadOnly()
```

Comment:
Is it worth to have method for a single call?

Reply:
Addressed in recent commits